stan/math/rev/functor/operands_and_partials.hpp

```diff
@@ -130,8 +133,20 @@ class operands_and_partials<Op1, Op2, Op3, Op4, Op5, var> {
     edge5_.dump_operands(&varis[idx += edge4_.size()]);
     edge5_.dump_partials(&partials[idx]);
 
+    auto container_operands = std::tuple_cat(
```

Comment:
The empty tuples have been bothering me too. If there's a memory leak, then I suspect this code will get touched in the rewrite.

In that case I like a function `collect_container_operands` that goes and looks in each edge and pulls out the right bits (so all the edges don't need extra functions).

Also I'm Pacific time zone and down to pair program this via Hangouts or whatever. I tried something myself and faceplanted.

Reply:
Hmm, there is 9-hour difference between pacific time and CEST. I guess I can take time for this one afternoon. Btw I prefere Skype if you are using it (I have some issues with VoiPs within browsers on PC, but I can use laptop).